bokehjs/test/integration/annotations/legend.ts

```diff
@@ -48,4 +48,236 @@ describe("Legend annotation", () => {
 
     await display(p)
   })
+
+  it(`should display title correctly`, async () => {
+    const p = fig([225, 125])
+    p.add_layout(new LinearAxis(), "above")
+    p.add_layout(new LinearAxis(), "right")
+
+    const x = 2
+    const y = 1
+    const cr = p.circle(x, y, {fill_color: "red"})
+
+    const items = [new LegendItem({label: "#0", renderers: [cr]})]
+
+    p.add_layout(new Legend({
+      items,
+      background_fill_alpha: 0.7,
+      location: "top_left",
+      orientation: "vertical",
```

Comment:
For now only vertical orientation is tested. Do the same for horizontal. That leads me to another point, which is this code needs to be deduplicated, and common core factored out into a function or a set of functions (see e.g. `integration/axes.ts`).

Reply:
Do you think it is necessary to test the combinatory locations, orientations and sides for each legend? something like 
```javascript
    const locations: Anchor[] = [
      "top_left", "top_center", "top_right",
      "center_left", "center_center", "center_right",
      "bottom_left", "bottom_center", "bottom_right",
      "top", "left", "center", "right", "bottom",
    ]

    const orientations: Orientation[] = [
      "horizontal", "vertical"
    ]

    const sides: Place[] = ["above", "below", "left", "right", "center"]

    locations.forEach(location => {
      orientations.forEach(orientation => {
        sides.forEach(side => {
          p.add_layout(legend({location, orientation}), side)
        })
      })
    })
```

Reply:
For now testing both orientations should be sufficient, as their logic is significantly different from each other, whereas anchors and locations (except for `center`) are implemented fairly uniformly, Especially anchors could be tested with unit tests.